activerecord/lib/active_record/connection_adapters/postgresql/referential_integrity.rb

```diff
@@ -45,8 +45,9 @@ def all_foreign_keys_valid? # :nodoc:
             BEGIN
             FOR r IN (
               SELECT FORMAT(
-                'UPDATE pg_constraint SET convalidated=false WHERE conname = ''%I''; ALTER TABLE %I VALIDATE CONSTRAINT %I;',
+                'UPDATE pg_constraint SET convalidated=false WHERE conname = ''%I''; ALTER TABLE %I.%I VALIDATE CONSTRAINT %I;',
```

Comment:
Would you also add a where clause to specify the schema name for `UPDATE pg_constrain` statements? It can make use of `connamespace::regnamespace`.

```
activerecord_unittest=# select conname, connamespace, contype, connamespace::regnamespace from pg_constraint  WHERE conname = 'fk_rails_94423a17a3' and connamespace::regnamespace = 'public'::regnamespace;
-[ RECORD 1 ]+--------------------
conname      | fk_rails_94423a17a3
connamespace | 2200
contype      | f
connamespace | public

activerecord_unittest=#
```

Reply:
Wouldn't this be a problem if there are ActiveRecord models (and foreign keys) in `public` and non-`public` schemas?

In that case we would only validate the ones in `public`, do we want to ignore the other ones?

Reply:
I mean to say this change that generates these SQL statements. I should have provided a complete example.

```diff
$ git diff
diff --git a/activerecord/lib/active_record/connection_adapters/postgresql/referential_integrity.rb b/activerecord/lib/active_record/connection_adapters/postgresql/referential_integrity.rb
index 2b87700bbb..747cd20ce1 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql/referential_integrity.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql/referential_integrity.rb
@@ -45,9 +45,10 @@ def all_foreign_keys_valid? # :nodoc:
             BEGIN
             FOR r IN (
               SELECT FORMAT(
-                'UPDATE pg_constraint SET convalidated=false WHERE conname = ''%I''; ALTER TABLE %I.%I VALIDATE CONSTRAINT %I;',
+                'UPDATE pg_constraint SET convalidated=false WHERE conname = ''%I'' and connamespace::regnamespace = ''%I''::regnamespace; ALTER TABLE %I.%I VALIDATE CONSTRAINT %I;',
                 constraint_name,
                 table_schema,
+                table_schema,
                 table_name,
                 constraint_name
               ) AS constraint_check
$
```

- Generated SQL statements after this change. 
```sql
UPDATE pg_constraint SET convalidated=false WHERE conname = 'fk_parent_node' and connamespace::regnamespace = 'referential_integrity_test_schema'::regnamespace; ALTER TABLE referential_integrity_test_schema.nodes VALIDATE CONSTRAINT fk_parent_node;
```

- Generated SQL statements without this change.
```sql
UPDATE pg_constraint SET convalidated=false WHERE conname = 'fk_parent_node'; ALTER TABLE referential_integrity_test_schema.nodes VALIDATE CONSTRAINT fk_parent_node;
```